### DISCUSSION TITLE
过期：个人的理论渲染效率排行

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#58] [回收]
# 换到此帖更新 https://github.com/hooke007/MPV_lazy/discussions/255#discussioncomment-4685339

### 如果使用opengl
一定要指定 `--gpu-context=win` https://mpv.io/manual/master/#options-gpu-context
不要用angle（d3d11转译效率打折、功能残缺，但它居然是官方 `gpu-api=opengl` 时的首选），不要用dxinterop（d3d9输出）

#### 一些特异点记录
- `--hwdec` 不支持 `d3d11va` 但支持 `nvdec` （auto时的优选）
- 如果使用ffmpeg，脚本缩略图无法生成显示

### 本机的阶段性结论
主要考虑性能 ——

`vo=gpu ; gpu-context` ：win10/11用户首选 `d3d11` （看实际情况选择是否关闭翻转），次选 `win` （没有做广泛的选项兼容测试）
`vo=gpu-next ; gpu-context` ：win10/11用户首选 `d3d11` （看实际情况选择是否关闭翻转），次选 `winvk` （没有做广泛的选项兼容测试）

## 跑分用的参数
i7 10875H + RTX 2070s lp （1080p h264 yuv420 9900+kbps sdr）
```
profile=BenchMark

[BenchMark]

vo=gpu # gpu-next
gpu-context=win # d3d11 winvk

sub=no
audio=no
keep-open=always
video-sync=display-desync
interpolation=no
vulkan-swap-mode=immediate # mailbox
opengl-swapinterval=0
opengl-dwmflush=no
d3d11-sync-interval=0
d3d11-flip=no # 翻转模型

hwdec=no
hidpi-window-scale=no
window-scale=0.7
cscale=jinc
cscale-radius=3.0
dscale=jinc
dscale-radius=3.0
#glsl-shaders="~~exe_dir/test.glsl"
osd-msg1="FPS: ${estimated-display-fps}"
```

**Comment by hooke007** [comment]
20220513 爆分是因为换了低码片源
|  FPS | vo | gpu-context |  | FPS | vo | gpu-context |
|  ---:  | ---  | ---  | --- | ---:  | ---  | ---  |
| 1750 | gpu | win | | 854? | gpu-next | win |
| 2106 | gpu | d3d11 | | 2603 | gpu-next | d3d11 |
| 358? | gpu | d3d11(flip) | | 358? | gpu-next | d3d11(flip) |
| 1189 | gpu | winvk | | 1094 | gpu-next | winvk |

gpu-next下的opengl依旧存在严重性能问题 https://github.com/mpv-player/mpv/issues/9551
需要关闭flip以完全解放d3d11的渲染性能 —— 显卡驱动？？？跑分垫底但尚未在实际播放中体现
vulkan的理论性能优势可能只适用于AMD家的显卡

edit: 换源之后差异被放大了。
gpu-next的og实际放4khdr的时候依旧是掉帧炮
d3d11下gpu-next比gpu更快已经做到了；flip这奇怪的限制表现的跟锁帧一样（瞎猜：前后两次测试，都是卡着我显示器刷新率数值的3倍上不去，三重缓冲？）
N卡的vulkan还是莫名弱不少

**Reply by dyphire** [reply]
我也测试了一下，n卡在d3d11的api下禁用flip在1080p上没有性能差距，但upscale至4K时是flip模式下的三倍左右。flip的性能问题应该是普遍存在于n卡驱动上了

**Comment by yeezylife** [comment]
我自己笨笨测试用软解性能最好...我这里软解3-4W码率的原盘,CPU占用百分之7左右,显卡温度从25度升高至26-7度...
我用其他任何硬解开这么多shaders去跑3-4W的码率,显卡都会升温十几二十度

**Reply by hooke007** [reply]
我这个帖测得是极限效率（不管是什么处理器，跑分模式下占用都是拉满的），不是功耗/温度，顺带测了测是否后端存在的bug
